In [3]:
import random 

random_list = [random.random() for _ in range(5_000_000)]

print("random list length:", len(random_list))

random list length: 5000000


In [4]:
def sd(nums):
    avg = sum(nums)/len(nums)

    summ = 0

    for i in range(len(nums)):
        val = (nums[i]-avg)**2

        summ += val

    var = summ/len(nums)

    return var**0.5

print("random list standard deviation:", sd(random_list))

random list standard deviation: 0.28861076697814697


In [5]:
import numpy as np 

arr = np.array(random_list)

print("random list standard deviation:", arr.std())

random list standard deviation: 0.2886107669781472


In [6]:
import pandas as pd 

data = pd.Series(random_list)

print("Environment configured successfully! (yay)")
print(data.describe())

Environment configured successfully! (yay)
count    5.000000e+06
mean     5.001532e-01
std      2.886108e-01
min      1.899811e-07
25%      2.502367e-01
50%      5.002652e-01
75%      7.499740e-01
max      9.999998e-01
dtype: float64


In [13]:
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

df.groupby('Sex')['Fare'].mean()



Sex
female    44.479818
male      25.523893
Name: Fare, dtype: float64

In [16]:
# Outliers
q1 = df['Fare'].quantile(0.25)
q3 = df['Fare'].quantile(0.75)
icr = q3 - q1

min_outlier = q1 - 1.5*icr
max_outlier = q3 + 1.5*icr

print(f"El outlier minimo en fare es de {min_outlier} y el maximo de {max_outlier}")

El outlier minimo en fare es de -26.724 y el maximo de 65.6344


# Análisis del Dataset

- Pregunta A: Observen la columna Fare (Tarifa del boleto). ¿Cuál es la diferencia matemática entre el percentil 75% (el tercer cuartil) y el valor Máximo (max)? Como ingenieros de software, ¿qué les dice esta enorme dispersión estadística sobre la homogeneidad de los precios que se cobraron en el barco?

Es muy grande, es de 481.32392, lo que nos indica que hay boletos que se vendieron por mucho más que otros. A su vez, podemos asumir que la media no podrá darnos una buena
interpretación de los valores de la tarifa.

- Pregunta B: Revisen la fila count (conteo total de registros válidos) de las distintas columnas en el resultado de su .describe(). Comparen el conteo de la columna PassengerId con el conteo de la columna Age. ¿Qué anomalía de base de datos acaban de descubrir?

Hay algunos valores de la columna Age que no están registrados en el dataset (son null).

- Pregunta C: El algoritmo que construirán al final del semestre requiere que todos los pasajeros tengan un valor numérico en la columna Age para poder compilar. Sabiendo lo que acaban de descubrir en la Pregunta B, propongan al menos dos estrategias a nivel de base de datos o de programación para resolver este problema antes de alimentar a la IA.

Podemos reemplazar los valores nulos en la columna Age con la mediana para mantener la distribución correcta de los datos. 

- ¿Cuál fue la tarifa promedio que pagaron las mujeres?

44.479818

- ¿Y cuál fue la de los hombres?

25.523893

In [17]:
print(df['Survived'].value_counts(normalize=True))

Survived
0    0.616162
1    0.383838
Name: proportion, dtype: float64
